src/bin/pg_autoctl/service_keeper.c

```diff
@@ -402,77 +372,12 @@ keeper_node_active_loop(Keeper *keeper, pid_t start_pid)
 
 		couldContactMonitor = couldContactMonitorThisRound;
 
-		if (couldContactMonitor)
-		{
-			char expectedSlotName[BUFSIZE];
-
-			keeperState->last_monitor_contact = now;
-			keeperState->assigned_role = assignedState.state;
-
-			if (keeperState->assigned_role != keeperState->current_role)
-			{
-				needStateChange = true;
-
-				log_info("Monitor assigned new state \"%s\"",
-						 NodeStateToString(keeperState->assigned_role));
-			}
-
-			/*
-			 * Also update the groupId and replication slot name in the
-			 * configuration file.
-			 */
-			(void) postgres_sprintf_replicationSlotName(assignedState.nodeId,
-														expectedSlotName,
-														sizeof(expectedSlotName));
-
-			if (assignedState.groupId != config->groupId ||
-				strneq(config->replication_slot_name, expectedSlotName))
-			{
-				bool postgresNotRunningIsOk = false;
-
-				if (!keeper_config_set_groupId_and_slot_name(config,
-															 assignedState.nodeId,
-															 assignedState.groupId))
-				{
-					log_error("Failed to update the configuration file "
-							  "with groupId %d and replication.slot \"%s\"",
-							  assignedState.groupId, expectedSlotName);
-					return false;
-				}
-
-				if (!keeper_ensure_configuration(keeper, postgresNotRunningIsOk))
-				{
-					log_error("Failed to update our Postgres configuration "
-							  "after a change of groupId or "
-							  "replication slot name, see above for details");
-					return false;
-				}
-			}
-		}
-		else
+		if (keeperState->assigned_role != keeperState->current_role)
 		{
-			log_error("Failed to get the goal state from the monitor");
+			needStateChange = true;
```

Comment:
There is a behavioural change here. When going to `DEMOTE_TIMEOUT_STATE`, we actually did not set `needStateChange` before. But we do now, because we set it regardless of `couldContactMonitor`.

Reply:
Fixed, thanks.